Frends.ServiceBus.Read/Frends.ServiceBus.Read/Definitions/MessagingFactory.cs

```diff
@@ -0,0 +1,147 @@
+    /// <param name="path"></param>
+    /// <param name="timeout"></param>
+    /// <returns></returns>
+    public MessageSender GetMessageSender(string connectionString, string path, TimeSpan timeout)
+    {
+        return new MessageSender(GetCachedMessagingFactory(connectionString, timeout), path);
+    }
+
+    private ServiceBusConnection GetCachedMessagingFactory(string connectionString, TimeSpan timeout)
+    {
+        string key = timeout.TotalSeconds + "-" + connectionString;
+
+        if (!_connections.ContainsKey(key))
+        {
+            lock (factoryLock) // TODO: change double check
+            {
+                if (!_connections.ContainsKey(key))
+                {
+                    _connections.TryAdd(key, CreateConnectionWithTimeout(connectionString, timeout));
+                }
+            }
+        }
+
+        return _connections[key];
+    }
+
+    /// <summary>
+    /// Create new client for servicebus connection. This method is slow!
+    /// </summary>
+    /// <param name="connectionString">Connection string</param>
+    /// <param name="operationTimeoutForClients">Operation timeout for clients</param>
+    /// <returns>Object that can handle messaging to the service bus</returns>
+    internal static ServiceBusConnection CreateConnectionWithTimeout(string connectionString, TimeSpan operationTimeoutForClients)
+    {
+        var connBuilder = new ServiceBusConnectionStringBuilder(connectionString)
+        {
+            OperationTimeout = operationTimeoutForClients
+        };
+
+        var connection = new ServiceBusConnection(connBuilder) { RetryPolicy = RetryPolicy.Default };
+
+        return connection;
+    }
+
+
+    #region IDisposable Support
+    private bool _disposedValue = false; // To detect redundant calls
+
+    private void Dispose(bool disposing)
+    {
+        if (!_disposedValue)
+        {
+            var factoriesToClose = _connections.ToList();
+            _connections.Clear();
+
+            if (disposing)
+            {
+                // TODO: dispose managed state (managed objects).
+            }
+
+            foreach (var item in factoriesToClose)
+            {
+                try
+                {
+                    item.Value.CloseAsync().Wait();
+                }
+                catch (Exception ex)
+                {
+                    Trace.TraceError("Error when aborting messaging factory connection " + ex);
+                }
+            }
+
+            _disposedValue = true;
+        }
+    }
+
+    // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
+    /// <summary>Allows an object to try to free resources and perform other cleanup operations before it is reclaimed by garbage collection.</summary>
+    //~ServiceBusMessagingFactory()
+    //{
+    //    // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+    //    //Dispose(false);
+    //}
```

Comment:
XML warnings should all be handled. I think this is because there is another <summary></summary> above the Dispose method.

Reply:
Fixed. ServiceBusMessagingFactory() is commented so there's nothing where XML could be linked and Dispose() got it's own XML as you mentioned.